File: backend/src/main/java/com/example/transportation_company/entity/Offer.java
package com.example.transportation_company.entity;

import com.example.transportation_company.enums.State;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "offers")
public class Offer {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long id;
    private String transportationInfo;
    private double price;
    @Enumerated(EnumType.STRING)
    private State state;

    @ManyToOne()
    private Transporter transporter;

    @ManyToOne()
    private Delivery delivery;

}

File: frontend/src/app/api/models/Offer.ts
import {State} from "./enums.ts/State";
import {Delivery} from "./Delivery";
// @ts-ignore
import {Transporter} from "./Transporter";

export interface Offer{
  id: number,
  transportationInfo: string,
  price: number,
  state: State,
  transporter: Transporter,
  delivery: Delivery

}
